src/xref2/component.ml

```diff
@@ -31,26 +31,6 @@ module IdentMap = Map.Make (struct
 
   let compare = Ident.compare
 end)
-    
-
-module Delayed = struct
-  let eager = ref false
-
-  type 'a t = { mutable v : 'a option; get : unit -> 'a }
-
-  let get : 'a t -> 'a =
-   fun x ->
-    match x.v with
-    | Some x -> x
-    | None ->
-        let v = x.get () in
-        x.v <- Some v;
-        v
-
-  let put : (unit -> 'a) -> 'a t =
-   fun f ->
-    if !eager then { v = Some (f ()); get = f } else { v = None; get = f }
-end
```

Comment:
I'm still not keen on moving this module to Subst - we'll be delaying strengthening soon too, for instance.

Reply:
This module is not moved, it's removed.
`Subst.Delayed` doesn't do delayed computation, it's the corresponding module for `'a Substitution.delayed`.

Reply:
Ah, I misread this as `Subst.delayed` - even so though, it's still within the `Substitution` module, which is going to be odd when we add in strengthening.

Reply:
By delaying strengthening, do you mean lazy computation or composed substitutions ?
In the first case, `Subst.Delayed` doesn't do that. I reused the name for something unrelated.
In the second, it is necessary to depend on `Subst`.

Reply:
I'm imagining the delayed stuff as a list of transforms that we're applying to a more-or-less immutable base. Substs are one such transformation, strengthening is another.

Reply:
If there is code/types to share between subst and strengthening, I think we should do it later.
For now, `Component.Delayed` is removed and `Subst.Delayed` don't replace it. Should I bring `Component.Delayed` back ?

Reply:
It's not so much that there is code to share as that conceptually we're moving in the right direction. In any case, it's something we can sort out later as you say. LGTM :-)